File: aula_5/aula_5.m

# Aula 5 - soluçao de equacoes de uma variavel

#Metodo grafico
#Plotar grafico
f = @(x) exp(x) -x - 2; #Define a funçao
%xx = linspace(-5, 5);   #Cria vetor com 100 valores no intervalo [-5, 5]
%yy = f(xx);
%plot(xx, yy);grid

#Metodo da bissecçao
a = 1;
b = 2;
x = (a + b) / 2;
[f(a) f(x) f(b)]

b = x;
x = (a + b) / 2;
[f(a) f(x) f(b)]

b = x;
x = (a + b) / 2;
[f(a) f(x) f(b)]

a = x;
x = (a + b) / 2;
[f(a) f(x) f(b)]

b = x;
x = (a + b) / 2;
[f(a) f(x) f(b)]

b = x;
x = (a + b) / 2;
[f(a) f(x) f(b)]

a = x;
x = (a + b) / 2;
[f(a) f(x) f(b)]

b = x;
x = (a + b) / 2;
[f(a) f(x) f(b)]

b = x;
x = (a + b) / 2;
[f(a) f(x) f(b)]

disp(x)

File: aula_5/aula_5.py
from math import exp
import matplotlib.pyplot as plt
import numpy as np

"""
    aula 5 - solução de equações de uma variável
"""

# Método gráfico
# Plotar gráfico
xx = np.arange(-5, 5, 0.1)  # Cria vetor com 100 valores no intervalo [-5, 5)
yy = list(map(lambda x: exp(x) - x - 2, xx))
fig, ax = plt.subplots()
ax.plot(xx, yy)
ax.grid()
# plt.show()


def f(x):
    return exp(x) - x - 2


# Método da bissecção
a = 1
b = 2
x = (a + b) / 2
print('{} {} {}'.format(f(a), f(x), f(b)))

b = x
x = (a + b) / 2
print('{} {} {}'.format(f(a), f(x), f(b)))

b = x
x = (a + b) / 2
print('{} {} {}'.format(f(a), f(x), f(b)))

a = x
x = (a + b) / 2
print('{} {} {}'.format(f(a), f(x), f(b)))

b = x
x = (a + b) / 2
print('{} {} {}'.format(f(a), f(x), f(b)))

b = x
x = (a + b) / 2
print('{} {} {}'.format(f(a), f(x), f(b)))

a = x
x = (a + b) / 2
print('{} {} {}'.format(f(a), f(x), f(b)))

b = x
x = (a + b) / 2
print('{} {} {}'.format(f(a), f(x), f(b)))

a = x
x = (a + b) / 2
print('{} {} {}'.format(f(a), f(x), f(b)))

b = x
x = (a + b) / 2
print('{} {} {}'.format(f(a), f(x), f(b)))

print("x = {}".format(x))
